src/libraries/System.Private.CoreLib/src/System/Globalization/TextInfo.WebAssembly.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Globalization
+{
+    internal static unsafe class TextInfoInterop
+    {
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern unsafe void ChangeCaseInvariantJS(out string exceptionMessage, char* src, int srcLen, char* dstBuffer, int dstBufferCapacity, bool bToUpper);
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern unsafe void ChangeCaseJS(out string exceptionMessage, in string culture, char* src, int srcLen, char* dstBuffer, int dstBufferCapacity, bool bToUpper);
+    }
+
+    public partial class TextInfo
+    {
+        internal unsafe void JsChangeCase(char* src, int srcLen, char* dstBuffer, int dstBufferCapacity, bool toUpper)
+        {
+            Debug.Assert(!GlobalizationMode.Invariant);
+            Debug.Assert(!GlobalizationMode.UseNls);
+            Debug.Assert(GlobalizationMode.Hybrid);
+
+            string exceptionMessage;
+            if (IsInvariant)
```

Comment:
Where is IsInvariant defined? This function will work for hybrid mode so GlobalizationMode.Invariant should be false always.

Reply:
Defined here:
https://github.com/dotnet/runtime/blob/57ae91cf6fc5672e34705b1a272cf268761d505a/src/libraries/System.Private.CoreLib/src/System/Globalization/TextInfo.Icu.cs#L19
It is not about the invariant mode set explicitly. I will leave it as it is, it's a safety measure in case `TextInfo` class would not have a access to valid culture name. Sending empty `_cultureName` to `ChangeCaseJS` does not make much sense.

Reply:
I will move the property to `TextInfo.cs` as now it's not exclusivly used by `.Icu.cs`, it will make it easier to find.